modules/vba_documents/app/sidekiq/vba_documents/upload_processor.rb

```diff
@@ -90,6 +94,38 @@ def original_file_metadata(tempfile)
       }
     end
 
+    def find_icn(file_number)
+      bgss = BGS::Services.new(external_uid: file_number, external_key: file_number)
+
+      # File Number is ssn, file number, or participant id.  Call BGS to get the
+      # veterans birthdate
+      # rubocop:disable Rails/DynamicFindBy
+      bgs_vet = bgss.people.find_by_ssn(file_number) || # rubocop:disable Rails/DynamicFindBy
+                bgss.people.find_by_file_number(file_number) ||
+                bgss.people.find_person_by_ptcpnt_id(file_number)
+      # rubocop:enable Rails/DynamicFindBy
+      return nil if bgs_vet.blank?
+
+      # Go after ICN in MPI
+      mpi = MPI::Service.new
+      r = mpi.find_profile_by_attributes(first_name: bgs_vet[:first_nm].to_s,
+                                         last_name: bgs_vet[:last_nm].to_s,
+                                         ssn: bgs_vet[:ssn_nbr].to_s,
+                                         birth_date: bgs_vet[:brthdy_dt].strftime('%Y-%m-%d'))
+      return nil if r.blank? || r.profile.blank?
+
+      r.profile.icn
+
+    # at this point ICN is not required when submitting to EMMS, so have wide
+    # exception handling, log and move on, any errors trying to get ICN should not stop us from submitting
+    # rubocop:disable Lint/RescueException
+    rescue Exception => e
```

Comment:
Standard Rails practice is to never `rescue Exception`, but rather rescue `StandardError`, assuming you can't rescue a more specific error type. You can accomplish this by removing `Exception` and doing `rescue => e` instead, which will rescue anything under `StandardError`.

Here's [a good write-up](https://www.honeybadger.io/blog/ruby-exception-vs-standarderror-whats-the-difference/) on why rescuing from `Exception` is a bad idea.

Reply:
thanks, nice reference fixed